rct229/rulesets/ashrae9012019/section19/section19rule2.py

```diff
@@ -0,0 +1,188 @@
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section19Rule2.FluidLoopRule(),
+            index_rmr="baseline",
+            id="19-2",
+            description="Baseline building plant capacities shall be based on coincident loads.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1.2.2",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.fluid_loops[*]",
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+
+        hvac_ids_b = find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].id",
+            rmi_b,
+        )
+
+        HW_fluid_loop_list = []
+        CW_fluid_loop_list = []
+        CHW_fluid_loop_list = []
+        for hvac_id_b in hvac_ids_b:
+            if is_hvac_sys_heating_type_fluid_loop(
+                rmi_b, hvac_id_b
+            ) and is_hvac_sys_fluid_loop_attached_to_boiler(rmi_b, hvac_id_b):
+                HW_fluid_loop_list.append(
+                    getattr_(
+                        find_exactly_one_hvac_system(rmi_b, hvac_id_b),
+                        "hot_water_loop",
+                        "heating_system",
+                        "hot_water_loop",
+                    )
+                )
+
+            if is_hvac_sys_preheating_type_fluid_loop(
+                rmi_b, hvac_id_b
+            ) and is_hvac_sys_preheat_fluid_loop_attached_to_boiler(rmi_b, hvac_id_b):
+                HW_fluid_loop_list.append(
+                    getattr_(
+                        find_exactly_one_hvac_system(rmi_b, hvac_id_b),
+                        "hot_water_loop",
+                        "preheat_system",
+                        "hot_water_loop",
+                    )
+                )
+
+            if is_hvac_sys_cooling_type_fluid_loop(
+                rmi_b, hvac_id_b
+            ) and is_hvac_sys_fluid_loop_attached_to_chiller(rmi_b, hvac_id_b):
+                CW_fluid_loop_list.append(
+                    getattr_(
+                        find_exactly_one_hvac_system(rmi_b, hvac_id_b),
+                        "chilled_water_loop",
+                        "cooling_system",
+                        "chilled_water_loop",
+                    )
+                )
+
+        for CHW_child_loop in find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].cooling_system.chilled_water_loop",
+            rmi_b,
+        ):
+            CHW_fluid_loop_list.append(CHW_child_loop)
+
+        # extend `HW_fluid_loop_list` with terminals
+        HW_fluid_loop_list.extend(
+            [
+                terminal["heating_from_loop"]
+                for terminal in find_all(
+                    "$.buildings[*].building_segments[*].zones[*].terminals[*]", rmi_b
+                )
```

Comment:
I suggest to move the `heating_source` logic to the `find_all`, since it is the only criteria:

```python
find_all(f'$.buildings[*].building_segments[*].zones[*].terminals[*][?(@.type="{HEATING_SOURCE.HOT_WATER}")]', rmi_b)
```

Reply:
Fixed.

Reply:
Fixed.